### ISSUE TITLE
Feedback on improving `unique` array API

### ISSUE THREAD

**Issue by jakirkham** [dask/dask#7853]
In the Data API meeting, we started discussing some pain points around [the current NumPy `unique` implementation]( https://numpy.org/doc/stable/reference/generated/numpy.unique.html ) and how we might improve this ( https://github.com/data-apis/array-api/issues/212 ). Part of the issue is this function tries to do a bunch of stuff, which results in some unpleasantness around implementation, usage, etc. Some ideas around how we might improve it our included in that issue. Would be curious to know how others think we might improve the API of `unique` generally as well as how this plays into use cases in practice.

**Comment by jakirkham** [comment]
Combing through the issues here have identified the following:

1. Doing tree reductions with `unique` is complicated due to all the return values provided ( https://github.com/dask/dask/issues/2851 )
2. Returning multiple arguments complicates things for Dask (making it hard to use CuPy or other things) ( https://github.com/dask/dask/issues/7482 )
3. Memory usage can be a problem for some additional arguments ( https://github.com/dask/dask/issues/6828 ) ( https://github.com/dask/dask/issues/5377 )

Admittedly with all of these there is some gray area in terms of whether these are issues with Dask itself, the API, or its usage.

**Comment by jakirkham** [comment]
@pentschev, would be good to get your thoughts here since we have discussed the oddities of `np.unique` before and the changes proposed in the Data API may help with that here

**Comment by pentschev** [comment]
@jakirkham yeah, I've seen this before, and I wrote https://github.com/dask/dask/issues/7997 covering two issues that involve structured arrays in Dask, one of them being `da.unique`. Unfortunately I haven't found any better ideas than using a `dict`, and in the case of `np.unique` I went ahead with https://github.com/dask/dask/pull/8021 to allow computing it with CuPy chunks, but for values only.

I'm happy to engage in further discussion, but I also got stuck with that problem without a good solution.

**Comment by jcrist** [comment]
Since this is mostly an upstream issue in Data API, I'm going to close this here and let discussion continue there.
